### PR TITLE
fix: 404 links issue

### LINKED ISSUE #9925: [CAL-2095] links on 404 broken
[https://cal.com/wasdasd](https://cal.com/wasdasd)

visitng a 404 page and then clicking docs will open a 404 for /docs

[Source image](https://github.com/calcom/cal.com/assets/8019099/82cd9c4d-8c15-4af8-b062-00a2f11be5e0)
[https://github.com/calcom/cal.com/assets/8019099/82cd9c4d-8c15-4af8-b062-00a2f11be5e0](https://github.com/calcom/cal.com/assets/8019099/82cd9c4d-8c15-4af8-b062-00a2f11be5e0)

clicking the first link Register ... wont pass the username

[Source image](https://github.com/calcom/cal.com/assets/8019099/5e33f36d-aa57-492d-adcc-aa0d335a9745)
[https://github.com/calcom/cal.com/assets/8019099/5e33f36d-aa57-492d-adcc-aa0d335a9745](https://github.com/calcom/cal.com/assets/8019099/5e33f36d-aa57-492d-adcc-aa0d335a9745)

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-2095](https://linear.app/calcom/issue/CAL-2095/links-on-404-broken)</sub>

### PR DESCRIPTION
**Pull request by leog**
## What does this PR do?

Links in 404 were not working, this fixes that

Fixes #9925 

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

Try going to a non-existent user such as `/asdffdsa` and see if links to register it and popular pages are working or not. Also see the community link and icon refer to Discord in 404 and in the app under the user dropdown.

### CODE REVIEW

**Comment by PeerRich** [comment]
this looks ready, no? can i merge?

**Comment by leog** [comment]
> this looks ready, no? can i merge?

Yes, tested in dev (cal-staging.com) now that it works according to the environments entry I created in our internal handbook

File: apps/web/pages/404.tsx [hunk 1]
```diff
@@ -58,24 +58,33 @@ export default function Custom404() {
...
 
   const isSuccessPage = router.asPath.startsWith("/booking");
   const isSubpage = router.asPath.includes("/", 2) || isSuccessPage;
   const isSignup = router.asPath.startsWith("/signup");
-  const isCalcom = process.env.NEXT_PUBLIC_WEBAPP_URL === "https://app.cal.com";
+  const isCalcom =
+    WEBAPP_URL &&
+    (new URL(WEBAPP_URL).hostname.endsWith("cal.com") ||
+      new URL(WEBAPP_URL).hostname.endsWith("cal.dev") ||
+      new URL(WEBAPP_URL).hostname.endsWith("cal-staging.com"));
```
Comment (PeerRich) [inline apps/web/pages/404.tsx:87]: @leog did you not move this into consts?

**Review by PeerRich — DISMISSED** [review]

File: apps/web/pages/404.tsx (hunk 1 above)
Comment (PeerRich) [reply apps/web/pages/404.tsx:87]: isCalcom is being replaced with IS_CALCOM right?

**Review by PeerRich — APPROVED** [review]

**Pull request merged**